local_planner/src/nodes/local_planner_nodelet.cpp

```diff
@@ -427,17 +426,24 @@ void LocalPlannerNodelet::printPointInfo(double x, double y, double z) {
 }
 
 void LocalPlannerNodelet::pointCloudCallback(const sensor_msgs::PointCloud2::ConstPtr& msg, int index) {
-  std::lock_guard<std::mutex> lck(*(cameras_[index].cloud_msg_mutex_));
-  cameras_[index].newest_cloud_msg_ = *msg;  // FIXME: avoid a copy
-  cameras_[index].received_ = true;
-  if (!cameras_[index].transform_registered_) {
-    std::pair<std::string, std::string> transform_frames;
-    transform_frames.first = msg->header.frame_id;
-    transform_frames.second = "/local_origin";
-    buffered_transforms_.push_back(transform_frames);
-    cameras_[index].transform_registered_ = true;
+  float time_since_last_cloud_update =
+      fabsf(cameras_[index].newest_cloud_msg_.header.stamp.toSec() - msg->header.stamp.toSec());
+  if (cameras_[index].transformed_ || time_since_last_cloud_update > 1.0) {
+    std::lock_guard<std::mutex> lck(*(cameras_[index].cloud_msg_mutex_));
```

Comment:
I think we need the mutex at the top, otherwise we could access the timestamp while it is in the middle of being updated and get a corrupted value.

Reply:
true. Fixed